app/src/battery.c

```diff
@@ -18,16 +19,20 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 #include <zmk/battery.h>
 #include <zmk/events/battery_state_changed.h>
 
-const struct device *battery;
-
 static uint8_t last_state_of_charge = 0;
 
 uint8_t zmk_battery_state_of_charge() { return last_state_of_charge; }
 
+#if DT_HAS_CHOSEN(zmk_battery)
+
+#define BATTERY_DEV_LABEL DT_LABEL(DT_CHOSEN(zmk_battery))
```

Comment:
We should be able to use some condition defines magic to fallback to the hardcoded label if a chosen node isn't found.

Ideally, we can also *warn* and add a deprecation notice, since there are quite a few out of tree boards that will break if we don't leave compatibility for a while.

Reply:
I can add a fallback, but I don't think I can make a compile time warning about deprecation. I could add a log warning (that probably nobody will ever see). I don't think it is possible find a node with a given label property at compile time, which means I can't identify at compile time whether you are using `label = "BATTERY"` or you just don't have a battery sensor.

Reply:
You should be able to do something like:

```
#if DT_HAS_CHOSEN{zmk_battery)
#define BATTERY_NODE_ID DT_CHOSEN(zmk_battery)
#else
#warn "Use of 'BATTERY' node label is deprecated, use `zmk,battery` chosen node instead."
#define BATTERY_NODE_ID DT_NODELABEL("BATTERY")
#endif
```

And then use the define in the rest of the code.

Reply:
Summarized discussion from Discord: DT_NODELABEL() is for the other kind of label. There is no way to get a node by its label property at compile time, so unfortunately this is the best we can do.